src/main/java/net/pms/dlna/DLNAThumbnail.java

```diff
@@ -0,0 +1,263 @@
+	 *         source image could not be parsed.
+	 */
+	public static DLNAThumbnail toThumbnail(InputStream inputStream) {
+		return toThumbnail(inputStream, 0, 0, null, ImageFormat.SOURCE);
+	}
+
+	/**
+	 * Converts an image to a {@link DLNAThumbnail}. Format support is limited
+	 * to that of {@link ImageIO}.
+	 *
+	 * <p><b>
+	 * This method consumes and closes {@code inputStream}.
+	 * </b>
+	 * @param inputStream the source image in a supported format.
+	 * @param width the new width or 0 to disable scaling.
+	 * @param height the new height or 0 to disable scaling.
+	 * @param scaleType the {@link ScaleType} to use when scaling.
+	 * @param outputFormat the {@link ImageFormat} to generate or
+	 *                     {@link ImageFormat#SOURCE} to preserve source format.
+	 * @return The populated {@link DLNAThumbnail} or {@code null} if the
+	 *         source image could not be parsed.
+	 */
+	public static DLNAThumbnail toThumbnail(InputStream inputStream, int width, int height, ScaleType scaleType, ImageFormat outputFormat) {
+		if (inputStream == null) {
+			return null;
+		}
+		try {
+			// A little optimization that saves us from recreating an image that
+			// is already in the correct format/size and is a byte array.
+			byte[] bytes = null;
+			if (inputStream instanceof ByteArrayInputStream) {
+				bytes = ImagesUtil.toByteArray(inputStream);
+				inputStream = new ByteArrayInputStream(bytes);
+			}
+
+			ImageReaderResult imageResult = CustomImageReader.read(inputStream);
+			inputStream.close();
+
+			if (imageResult.bufferedImage == null || imageResult.imageFormat == null) { // ImageIO doesn't support the image format
+				LOGGER.error("Failed to create thumbnail because the source format is unknown");
+				return null;
+			}
+
+			if (outputFormat == null || outputFormat.equals(ImageFormat.SOURCE)) {
+				outputFormat = imageResult.imageFormat;
+			}
+
+			// Thumbnails aren't useful in all formats, restrict them
+			if (!ImageFormat.JPEG.equals(outputFormat) && !ImageFormat.PNG.equals(outputFormat)) {
+				outputFormat = ImageFormat.JPEG;
+			}
+
+			if (width == 0 ||
+				height == 0 ||
+				(
+					ScaleType.MAX.equals(scaleType) &&
+					imageResult.bufferedImage.getWidth() <= width &&
+					imageResult.bufferedImage.getHeight() <= height
+				)
+			) {
+				//No resize, just convert
+				if (imageResult.imageFormat.equals(outputFormat) && bytes != null) {
+					// Return source
+					return new DLNAThumbnail(bytes, imageResult.bufferedImage.getWidth(), imageResult.bufferedImage.getHeight(), imageResult.imageFormat, false);
+				}
+				ByteArrayOutputStream out = new ByteArrayOutputStream();
+				ImageIO.setUseCache(false);
+				ImagesUtil.imageIOWrite(imageResult.bufferedImage, outputFormat.toString(), out);
+				return new DLNAThumbnail(out.toByteArray(), imageResult.bufferedImage.getWidth(), imageResult.bufferedImage.getHeight(), outputFormat, false);
+			} else {
+				BufferedImage bufferedImage =
```

Comment:
@Nadahar It was a stupid question, as i forget that you told me that your IDE tell you when it's wrong or that something missing...
But whatever, i learned something more from you again :wink: 
Yes, my question was referring to your first answer, as i was not aware about this.